script/pgsqlms

```diff
@@ -74,6 +74,186 @@ my $ATTRD_PRIV    = "$HA_SBIN_DIR/attrd_updater --private --lifetime reboot";
+    my $tmpfile;
+    my @res;
+    my $ans;
+    my $pid;
+    my $rc;
+
+    unless ( defined $res and defined $query and $query ne '' ) {
+        ocf_log( 'debug', '_query: wrong parameters!' );
+        return -1;
+    }
+
+    unless ( $tmpfile = File::Temp->new(
+            TEMPLATE => 'pgsqlms-XXXXXXXX',
+            DIR      => $HA_RSCTMP
+        ) )
+    {
+        ocf_exit_reason( 'Could not create or write in a temp file' );
+        exit $OCF_ERR_INSTALLED;
+    }
+
+    print $tmpfile $query;
+    chmod 0644, $tmpfile;
+
+    ocf_log( 'debug', '_query: %s', $query );
+
+    # Change the effective user to the given system_user so after forking
+    # the given uid to the process should allow psql to connect w/o password
+    $> = $postgres_uid;
+
+    # Forking + piping
+    $pid = open(my $KID, "-|");
+
+    if ( $pid == 0 ) { # child
+        exec $PGPSQL, '--set', 'ON_ERROR_STOP=1', '-qXAtf', $tmpfile,
+            '-R', $RS, '-F', $FS, '--port', $pgport, '--host', $pghost,
+            $connstr;
+    }
+
+    # parent
+    $> = $oldeuid;
+
+    {
+        local $/;
+        $ans = <$KID>;
+    }
+
+    close $KID;
+    $rc = $? >> 8;
+
+    ocf_log( 'debug', '_query: psql return code: %d', $rc );
+
+    if ( defined $ans ) {
+        chop $ans;
+
+        push @{ $res }, [ split(chr(3) => $_, -1) ]
+            foreach split (chr(30) => $ans, -1);
+
+        ocf_log( 'debug', '_query: @res: %s',
+            Data::Dumper->new( [ $res ] )->Terse(1)->Dump );
+    }
+
+    # Possible return codes:
+    #  -1: wrong parameters
+    #   0: OK
+    #   1: failed to get resources (memory, missing file, ...)
+    #   2: unable to connect
+    #   3: query failed
+    return $rc;
+}
+
+# Get the last received location on a standby
+# if the first argument is true, returns the value as decimal
+# if the first argument is false, returns the value as LSN
+# Returns undef if query failed
+sub _get_last_received_lsn {
+    my ( $dec ) = @_;
+    my $pg_last_wal_receive_lsn = 'pg_last_wal_receive_lsn()';
+    my $pg_wal_lsn_diff         = 'pg_wal_lsn_diff';
+    my $query;
+    my $rc;
+    my @rs;
+
+    if ( $PGVERNUM < $PGVER_10  ) {
+        $pg_last_wal_receive_lsn = 'pg_last_xlog_receive_location()';
+        $pg_wal_lsn_diff         = 'pg_xlog_location_diff';
+    }
+
+    if ( $dec ) {
+        $query = "SELECT $pg_wal_lsn_diff( $pg_last_wal_receive_lsn, '0/0' )";
+    }
+    else {
+        $query = "SELECT $pg_last_wal_receive_lsn";
+    }
+
+    $rc = _query( $query, \@rs );
+
+    return $rs[0][0] if $rc == 0 and $rs[0][0];
+
+    ocf_log( 'err', 'Could not query last received LSN (%s)', $rc ) if $rc != 0;
+    ocf_log( 'err', 'No values for last received LSN' )
+        if $rc == 0 and not $rs[0][0];
+
+    return undef;
+}
+
+# Get the master score for each connected standby
+# Returns directly the result set of the query or exit with an error.
+# Exits with OCF_ERR_GENERIC if the query failed
+sub _get_lag_scores {
```

Comment:
Same here, I created this sub tho "hide" the mess with pg10 compat around the query from the callers.